### PR TITLE
[Layout Transition] Add default fade in / out layout transition

### PR DESCRIPTION
**Pull request by maicki**
Provide default fade & scale transition for layout transition API

@levi As discussed over at #2049 this PR implements a simple solution for now. cc @Adlai-Holler @appleguy

### CODE REVIEW

**Comment by Adlai-Holler** [comment]
@maicki This diff looks good to me. Could you add a simple example app that demonstrates the default transition behavior so we can easily test and modify this code?

**Comment by maicki** [comment]
@Adlai-Holler Added an example project: ASDKLayoutTransition

**Comment by Adlai-Holler** [comment]
This is great! Let's rename the proper to `defaultLayoutTransitionDuration` and update the comments to say this is the value we will use in the default code. That way users don't mistakenly think that they need to set/read this value in their custom transition implementation.

Once that's in, let's land this!

File: AsyncDisplayKit/ASDisplayNode.mm [hunk 1]
```diff
@@ -857,14 +859,93 @@ - (BOOL)_shouldAbortTransitionWithID:(int32_t)transitionID
...
+      }
+    }
+  }
+  
+  for (ASDisplayNode *insertedSubnode in insertedSubnodes) {
+    insertedSubnode.frame = [context finalFrameForNode:insertedSubnode];
+    insertedSubnode.alpha = 0;
+  }
+  
+  [UIView animateWithDuration:self.layoutTransitionDuration animations:^{
```
Comment (levi) [inline AsyncDisplayKit/ASDisplayNode.mm]: Expose options, duration & delay.

File: AsyncDisplayKit/ASDisplayNode.mm [hunk 2]
```diff
@@ -291,6 +291,10 @@ - (void)_initializeInstance
   
   _environmentState = ASEnvironmentStateMakeDefault();
   
+  _defaultLayoutTransitionDuration = 0.2;
+  _defaultLayoutTransitionDelay = 0.0;
+  _defaultLayoutTransitionOptions = NULL;
```
Comment (Adlai-Holler) [inline AsyncDisplayKit/ASDisplayNode.mm]: nit: Use `kNilOptions` here, since NULL is for pointers

Comment (Adlai-Holler) [inline AsyncDisplayKit/ASDisplayNode.mm]: Better yet, let's use `UIViewAnimationOptionBeginFromCurrentState` since that's most often what users want.

**Comment by Adlai-Holler** [comment]
Good to merge at will @maicki

File: AsyncDisplayKit/ASDisplayNode.mm (hunk 2 above)
Comment (Adlai-Holler) [inline AsyncDisplayKit/ASDisplayNode.mm]: It is indeed

Comment (maicki) [inline AsyncDisplayKit/ASDisplayNode.mm]: Let's add `kNilOptions` and `UIViewAnimationOptionBeginFromCurrentState` explicitly in here

**Comment by maicki** [comment]
@Adlai-Holler @levi  Good point with the layout options. Updated!

**Comment by Adlai-Holler** [comment]
Sweet!

**Pull request merged**